presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestUnion.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.sql.planner.LogicalPlanner;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.assertions.BasePlanTest;
+import com.facebook.presto.sql.planner.plan.AggregationNode;
+import com.facebook.presto.sql.planner.plan.ExchangeNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.facebook.presto.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
+import static com.facebook.presto.sql.planner.plan.ExchangeNode.Scope.REMOTE;
+import static java.util.stream.Collectors.toList;
+import static org.testng.Assert.assertTrue;
+
+public class TestUnion
+        extends BasePlanTest
+{
+    @Test
+    public void testPartialAggregationsWithUnion()
+    {
+        Plan plan = plan(
+                "SELECT orderstatus, sum(orderkey) FROM (SELECT orderkey, orderstatus FROM orders UNION ALL SELECT orderkey, orderstatus FROM orders) x GROUP BY (orderstatus)",
+                LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED,
+                false);
+        assertAtMostOneAggregationBetweenRemoteExchanges(plan);
+    }
+
+    private static void assertAtMostOneAggregationBetweenRemoteExchanges(Plan plan)
+    {
+        List<PlanNode> fragments = searchFrom(plan.getRoot())
+                .where(ExchangeNode.class::isInstance)
```

Comment:
if the `isRemoteExchange` would do what it tells it does then you could use it here.